.github/workflows/deploy-ghpages-latest-stable.yml

```diff
@@ -0,0 +1,151 @@
+    steps:
+      - name: checks for the label 
+        id: valid_label
+        run: |
+          if [[ "${{ inputs.trigger-label }}" == "latest" ]] || [[ "${{ inputs.trigger-label  }}" == "stable" ]];
+            then
+                echo "value=true" >> $GITHUB_OUTPUT
+            else
+                echo "value=false" >> $GITHUB_OUTPUT 
+          fi
+            
+  
+  build:
+    
+    runs-on: ubuntu-latest
+    needs: check-label
+    if: needs.check-label.outputs.label == 'true'
+    steps:
+      
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+        if: ${{ inputs.package-manager == 'poetry' }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+          cache: ${{ inputs.package-manager }}
+          cache-dependency-path: ${{ inputs.dependency-path }} 
+      - name: Install dependencies with pip
+        run: |
+          ${{ inputs.package-manager }} install .[docs]
+          # required by nbsphinx
+          # https://nbsphinx.readthedocs.io/en/0.2.15/installation.html#pandoc
+          sudo apt install pandoc
+        if : ${{ inputs.package-manager == 'pip' }}
+      - name: Install dependencies and compile the documentation with poetry
+        run: |
+          poetry install --no-interaction ${{ inputs.poetry-extras }}
+          pip install poethepoet
+          poe docs
+        if: ${{ inputs.package-manager == 'poetry' }}
+      - name: Compile documentation
+        working-directory: ${{ inputs.path-doc }}
+        run: |
+          make html
+        if : ${{ inputs.package-manager == 'pip' }} 
+      - name: Upload artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: documentation
+          path: ${{ inputs.path-doc }}/build/html
+  
+  # TODO: this job should have a own reusable workflow
+  new-artifact:
+    needs: [build, check-label]
+    if: needs.check-label.outputs.label == 'true'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2 
+    - name: generate the ${{inputs.trigger-label}} documentation 
+      run: |
+        label=${{inputs.trigger-label}}
+        mkdir new_docs
+        cd new_docs
+        mkdir $label
+    - uses: actions/download-artifact@v3
+      with:
+        name: documentation
+        path: new_docs/${{inputs.trigger-label}} 
+    - name: generate the remaning documentation
+      run: | 
+        label=${{inputs.trigger-label}} 
+        cd new_docs
+        if [[ ${label} == "latest" ]]; 
+          then 
+            missing_doc="stable"
+          else
+            missing_doc="latest"
+        fi
+        URL="https://qibogang.github.io/qibo/${missing_doc}/${missing_doc}"
+        mkdir "$missing_doc"
+        # Check if it is possible to connect to the website 
+        # Otherwise it copies the existing docs
+        if wget --spider "${URL}" 2>/dev/null; then 
```

Comment:
Why do you need `--spider`?

Usually, a spider is a program that follow links and download other pages. Here you are supposed to download exactly one thing (and most likely you are doing it, since the tar is not an HTML page, so there is no link to follow).
I guess is called "spider" because traverses the "net".